gsoc-2025/perf_reg_test.md

```diff
@@ -0,0 +1,32 @@
+---
+parent: DLang GSoC 2025 Project Ideas
+nav_order: 1
+---
+
+# Performance regression publisher
+
+**Mentor:** [Dennis Korpel](dkorpel@gmail.com)
+
+| Spec | Details |
+|-|-|
+| **Difficulty Level** | easy-medium |
+| **Project Duration** | 150 hours |
+| **Number of Contributors** | 1 |
+| **Prerequisites** | Github actions, webdev, performance testing |
+
+## Description
+
+The D compiler currently does not have an automated performance regression test. Oftentimes pull requests that claim to improve compiler performance are being made (be it spatial or temporal) however, it's up to the reviewer to actually believe the committer or to test things on their own. To make things simpler and more transparent we want to implement a bot that monitores all the pull requests that are being made to the compiler codebase and analyze the compiler performance with and without the pull request.
+
+The list of stats should include, but not be limited to:
+
+- size of some predefined binaries (like a "hello world" program)
+- compile time of popular projects
+- compiler size
+- runtime of test suite
+
+Ideally the bot could also store a history of performance regressions within a web page.
```

Comment:
You can be more specific here about how to achieve the bullets above. Maybe there are already existing open-source actions / bots that do this. @dkorpel do you know of any? If so, add links to them, or similar actions from other repos as models.

Reply:
The thing is, nothing is set in stone yet. I would let the student do some research on this.